src/app/main/main.component.ts

```diff
@@ -210,18 +215,24 @@ export class MainComponent implements OnInit, OnDestroy {
         });
 
       if (this._hasJobParams() && this._jobParamsValid()) {
+        this.dataFormGroup.patchValue({
+          sql: '/* Loading sql query from job... */',
+          projectID: this.project,
+          location: this.location
+        });
         this.dataService.getQueryFromJob(this.jobID, this.location, this.project).then((queryText) => {
           this.queryTextFromJob = queryText.sql;
           this.dataFormGroup.patchValue({
             sql: this.queryTextFromJob,
-            projectID: this.project
+            projectID: this.project,
+            location: this.location
           });
         });
       } else if (this._hasTableParams() && this._tableParamsValid()) {
         this.dataFormGroup.patchValue({
           sql: `SELECT * FROM \`${this.project}.${this.dataset}.${this.table}\`;`,
           projectID: this.project,
-          location: this.location
+          location: ''
```

Comment:
Do we need to clear `location`, or could we just not patch it here?

Reply:
Not patching it here is better.